File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/controller/specificationController.js
// controller 只和html交互 不直接和后台交互
app.controller("specificationController",function($scope,$controller, specificationService){  //直接把specificationService注入到controller中

    // 控制器的继承
      $controller('baseController',{$scope:$scope});//继承   本质上是共用同一个$scope

    // 动态添加行数据  本质就是向 entity.tbSpecificationOptionList中追加空对象
    // 注意 需要提前定义tbSpecificationOptionList属性  新建时ng-click="entity={tbSpecificationOptionList:[]}"
      $scope.addTbSpecificationOptionList=function () {
          $scope.entity.tbSpecificationOptionList.push({});
      }
// 动态删除行数据
    $scope.deleTbSpecificationOptionList=function (index) {
         $scope.entity.tbSpecificationOptionList.splice(index,1);
    }


        $scope.findPage=function(pageNum,pageSize){
            specificationService.findPage(pageNum,pageSize).success(function (response) {
                // 返回的数据 ：总条数 和 当前页的数据 {total:110,rows:[{},{},{}]}
                $scope.paginationConf.totalItems = response.total;
                $scope.list = response.rows;
            })
        }

        $scope.findAll=function () {
            specificationService.findAll().success(function(response){
                // response就是查询出的所有数据
                $scope.list = response;
            })
        }

        $scope.save=function () {
            var  obj = null;
            // entity.tbSpecification.id 是组合类
            if($scope.entity.tbSpecification.id!=null){
                obj = specificationService.update($scope.entity);
            }else{
                obj = specificationService.add($scope.entity);
            }
            obj.success(function (response) {
                if(response.success){
                    $scope.reloadList();
                }else{
                    alert(response.message);
                }
            });
        }

        $scope.findOne=function(id){
            specificationService.findOne(id).success(function (response) {
                // response就应该是返回的一个品牌对象
                $scope.entity=response;
            })
        }


        $scope.dele=function(){
            // 判断数组是否为空 如果为空 return
            if($scope.selectedIds.length==0){
                return;
            }

            var flag = window.confirm("确认要删除您选择的数据吗?");
            if(flag){
                specificationService.dele($scope.selectedIds).success(function (response) {
                    if(response.success){
                        $scope.reloadList();
                        // 把数组清空
                        $scope.selectedIds=[];
                    }else{
                        alert(response.message);
                    }
                })
            }
        }
        $scope.searchEntity={};//需要初始化 不然是undefined
        $scope.search=function(pageNum,pageSize){
            // 向后台传的参数有：$scope.searchEntity
            // 当前页码$scope.paginationConf.currentPage,每页显示的条数$scope.paginationConf.itemsPerPage
            specificationService.search(pageNum,pageSize,$scope.searchEntity).success(function (response) {
                // 返回的数据 ：总条数 和 当前页的数据 {total:110,rows:[{},{},{}]}
                $scope.paginationConf.totalItems = response.total;
                $scope.list = response.rows;
            })

        }
    })

File: pinyougou_parent/pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/SpecificationController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.sellergoods.service.SpecificationService;
import entity.PageResult;
import entity.Result;
import groupEntity.Specification;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

//@Controller
//@ResponseBody //转json 并且回显到浏览器
@RestController
@RequestMapping("/specification")
public class SpecificationController {

    @Reference
    private SpecificationService specificationService;

    @RequestMapping("/findSpecList")
    public List<Map> findSpecList(){
       return specificationService.findSpecList();
    }
    @RequestMapping("/findAll")
    public List<TbSpecification> findAll(){
       return specificationService.findAll();
    }
    @RequestMapping("/findPage")
    public PageResult findPage(int pageNum, int pageSize){
//        {total:110,rows:[{},{},{}]}
       return specificationService.findPage(pageNum,pageSize);
    }
    @RequestMapping("/add")  //组合类
    public Result add(@RequestBody Specification specification){
//        {success:true|false,message:"保存成功"|"保存失败"}
       try {
           specificationService.add(specification);
           return new Result(true,"保存成功");
       }catch (Exception e){
           e.printStackTrace();
           return new Result(false,"保存失败");
       }
    }

    @RequestMapping("/update")
    public Result update(@RequestBody Specification specification){
        try {
            specificationService.update(specification);
            return new Result(true,"修改成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }
    @RequestMapping("/dele")
    public Result dele(Long[] ids){
        try {
            specificationService.dele(ids);
            return new Result(true,"删除成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }
    @RequestMapping("/findOne")
    public Specification findOne(Long id){
        return  specificationService.findOne(id);
    }

    @RequestMapping("/search")
    public PageResult findPage(int pageNum, int pageSize,@RequestBody TbSpecification specification){
//        {total:110,rows:[{},{},{}]}
        return specificationService.search(pageNum,pageSize,specification);
    }

}
